### PR TITLE
#1577:  allow logging protobuf entries

### LINKED ISSUE #1577: Where are the allowed log entry protobuf types documented?
The [`LogEntry` API docs](https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/LogEntry#FIELDS.proto_payload) say:

    The log entry payload, represented as a protocol buffer. You can only use protoPayload values that belong to a set of approved types.

    An object containing fields of an arbitrary type. An additional field "@type" contains a URI identifying the type. Example: { "id": 1234, "@type": "types.example.com/standard/id" }.

Where are the allowed protobuf types documented, and the corresponding URIs?

### PR DESCRIPTION
**Pull request by tseaver**
Also, add a helper to parse an entry's payload into a protobuf message object.

Closes #1577

### CODE REVIEW

**Comment by tseaver** [comment]
Python 3.4 test failure is due to https://github.com/google/protobuf/issues/1352.

I will work around by using a different message type.

**Comment by tseaver** [comment]
@dhermes, @jgeewax PTAL

File: gcloud/logging/entries.py [hunk 1]
```diff
@@ -100,3 +104,11 @@ class ProtobufEntry(_BaseEntry):
...
     """
     _PAYLOAD_KEY = 'protoPayload'
+
+    def parse_message(self, message):
+        """Parse payload into a protobuf message.
+
+        :type message: Protobuf message
+        :param message: the message to be logged
+        """
+        Parse(json.dumps(self.payload), message)
```
Comment (dhermes) [inline gcloud/logging/entries.py]: No return type needed?

File: gcloud/logging/logger.py [hunk 2]
```diff
@@ -115,6 +119,35 @@ def log_struct(self, info, client=None):
...
+        :type message: Protobuf message
+        :param message: the message to be logged
+
+        :type client: :class:`gcloud.logging.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current logger.
+        """
+        client = self._require_client(client)
+        as_json_str = MessageToJson(message)
+        as_json = json.loads(as_json_str)
```
Comment (dhermes) [inline gcloud/logging/logger.py]: Hurts to have to `json.loads` only to `json.dumps` shortly after.

File: gcloud/logging/entries.py (hunk 1 above)
Comment (tseaver) [inline gcloud/logging/entries.py]: I couldn't see a need:  we are mutating the passed-in `message` in place.

File: gcloud/logging/logger.py (hunk 2 above)
Comment (tseaver) [inline gcloud/logging/logger.py]: Yup.  See https://github.com/google/protobuf/issues/1351

File: gcloud/logging/entries.py (hunk 1 above)
Comment (dhermes) [inline gcloud/logging/entries.py]: Gotcha. Can you say that in the docstring?

**Comment by dhermes** [comment]
LGTM pending docstring update

File: gcloud/logging/entries.py (hunk 1 above)
Comment (tseaver) [inline gcloud/logging/entries.py]: e9a01ea

**Pull request merged**